src/newgrf_station.h

```diff
@@ -79,13 +79,12 @@ struct StationResolverObject : public ResolverObject {
 	const SpriteGroup *ResolveReal(const RealSpriteGroup *group) const override;
 };
 
-enum StationClassID {
+enum StationClassID : byte {
 	STAT_CLASS_BEGIN = 0,    ///< the lowest valid value
 	STAT_CLASS_DFLT = 0,     ///< Default station class.
 	STAT_CLASS_WAYP,         ///< Waypoint class.
-	STAT_CLASS_MAX = 256,    ///< Maximum number of classes.
+	STAT_CLASS_MAX = 255,    ///< Maximum number of classes.
```

Comment:
This change may need more work because the max value is used as array declaration size in https://github.com/OpenTTD/OpenTTD/blob/66cd32a252ee0edab11448b560371878b2189223/src/newgrf_class.h#L32 and https://github.com/OpenTTD/OpenTTD/blob/66cd32a252ee0edab11448b560371878b2189223/src/newgrf_class_func.h#L26